website/project/model.py

```diff
@@ -3039,3 +3034,116 @@ def approve_embargo(self, user, token):
                 },
                 auth=Auth(user),
             )
+
+class DraftRegistration(AddonModelMixin, StoredObject):
+
+    is_draft_registration = True
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    datetime_initiated = fields.DateTimeField(auto_now_add=True)
+    datetime_updated = fields.DateTimeField(auto_now=True)
+
+    branched_from = fields.ForeignField('node')
+
+    initiator = fields.ForeignField('user')
+
+    registration_metadata = fields.DictionaryField({})
+    registration_schema = fields.ForeignField('metaschema')
+    registered_node = fields.ForeignField('node')
+
+    storage = fields.ForeignField('osfstoragenodesettings')
+
+    config = fields.DictionaryField()
+    flags = fields.DictionaryField()
+
+    def __init__(self, *args, **kwargs):
+        super(DraftRegistration, self).__init__(*args, **kwargs)
+
+        meta_schema = self.registration_schema or kwargs.get('registration_schema')
+        if meta_schema:
+            schema = meta_schema.schema
+            config = schema.get('config', {})
+            self.config = config
+            flags = schema.get('flags', {})
+            for flag, value in flags.iteritems():
+                self.flags[flag] = value
+
+    # proxy fields from branched_from Node
+    def __getattr__(self, attr):
+        try:
+            return self.__dict__[attr]
+        except KeyError:
+            return getattr(self.branched_from, attr, None)
+
+    def find_question(self, qid):
+        for page in self.registration_schema.schema['pages']:
+            for question_id, question in page['questions'].iteritems():
+                if question_id == qid and 'description' in question:
+                    return question['description']
```

Comment:
@barbour-em Why return just the question description? Let's return the whole thing.